### PR TITLE
[sso] Validate Issuer URL + fix trailing slash WEB-345

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

Fixes #

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - mp-sso-tra7454b9763d</li>
	<li><b>🔗 URL</b> - <a href="https://mp-sso-tra7454b9763d.preview.gitpod-dev.com/workspaces" target="_blank">mp-sso-tra7454b9763d.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [x] with-dedicated-emulation
- [x] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/public-api-server/pkg/oidc/oauth2.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func GetOAuth2ResultFromContext(ctx context.Context) *OAuth2Result {
 
 func (s *Service) OAuth2Middleware(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
-		config, _, err := s.GetClientConfigFromCallbackRequest(r)
+		config, _, err := s.getClientConfigFromCallbackRequest(r)
```
Comment (easyCZ) [inline components/public-api-server/pkg/oidc/oauth2.go:45]: drive-by change to make these methods private

**Review by AlexTugarev — APPROVED** [review]
Changes look good to me!
Would be nice to check with a preview env 🤞🏻

**Comment by easyCZ** [comment]
> Would be nice to check with a preview env 🤞🏻

It's currently not used anywhere, the follow-up PR will actually introduce it into the Sign-in dance

File: components/public-api-server/pkg/oidc/router_test.go [hunk 2]
```diff
@@ -60,6 +60,7 @@ func TestRoute_start(t *testing.T) {
 }
 
 func TestRoute_callback(t *testing.T) {
+	t.Skip()
```
[the remark this answers is not in the document]
Comment (easyCZ) [reply components/public-api-server/pkg/oidc/router_test.go]: I've for now removed the un-skip to land the fix, feel free to go ahead with the fix (I was planning to circle back to it after we land this fix)

**Comment by easyCZ** [comment]
I can confirm this works as a fix

Screenshot 2023-05-16 at 10 06 48
Screenshot 2023-05-16 at 10 06 09
